File: packages/java/cnos-java/src/main/java/ai/kitsy/cnos/CnosError.java
package ai.kitsy.cnos;

/**
 * Base exception for all CNOS runtime errors.
 */
public class CnosError extends RuntimeException {

    /** Thrown when no server projection can be located. */
    public static final String PROJECTION_NOT_FOUND = "cnos: no server projection found";

    /** Thrown when a required config key is absent. */
    public static final String MISSING_KEY = "cnos: missing config key";

    public CnosError(String message) {
        super(message);
    }

    public CnosError(String message, Throwable cause) {
        super(message, cause);
    }

    /** Returns true when the error signals that no projection was found. */
    public boolean isProjectionNotFound() {
        return getMessage() != null && getMessage().startsWith(PROJECTION_NOT_FOUND);
    }
}

File: packages/kotlin/cnos-kotlin/src/main/kotlin/ai/kitsy/cnos/CnosError.kt
package ai.kitsy.cnos

class CnosError(message: String, cause: Throwable? = null) : Exception(message, cause) {
    val isProjectionNotFound: Boolean
        get() = message?.startsWith("cnos: no projection") == true

    companion object {
        const val PROJECTION_NOT_FOUND = "cnos: no projection found"
        const val MISSING_KEY = "cnos: missing required config key"
    }
}
